File: dev-club/bcw3/Java/Basic/Vector/Vector.java
public class Vector {
    private double x;
    private double y;

    public Vector(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public Vector() {
        this.x = 0;
        this.y = 0;
    }

    public double getX() {
        return this.x;
    }

    public double getY() {
        return this.y;
    }

    public void setX(double value) {
        this.x = value;
    }

    public void setY(double value) {
        this.y = value;
    }

    public double len() {
        return Math.hypot(this.x, this.y);
    }

    public boolean equals(Vector other) {
        return this.x == other.x && this.y == other.y;
    }

    public boolean notequals(Vector other) {
        return !(this == other);
    }

    public void increment(Vector other) {
        this.x += other.x;
        this.y += other.y;
    }

    public void decrement(Vector other) {
        this.x -= other.x;
        this.y -= other.y;
    }

    public Vector sum(Vector other) {
        Vector result = this;

        result.x += other.x;
        result.y += other.y;

        return result;
    }

    public Vector substraction(Vector other) {
        Vector result = this;
    
        result.x -= other.x;
        result.y -= other.y;

        return result;    
    }
    
    public String toString() { 
        return "(" + this.x + ", " + this.y + ")"; 
    }
}

File: dev-club/bcw4/PHP/Vector/Vector.php
<?php  

class Vector {
    private $x;
    private $y;

    public function Vector($x, $y) {
        $this->x = $x;
        $this->y = $y;
    }

    public function getX() {
        return $this->x;
    }

    public function getY() {
        return $this->y;
    }

    public function setX($value) {
        $this->x = $x;
    }

    public function setY($value) {
        $this->y = $y;
    }

    public function len() {
        return hypot($this->x, $this->y);
    }

    public function compare($other) {
        return $this->x == $other->x && $this->y == $other->y;
    }

    public function isNotEqual($other) {
        return !($this == $other);
    }

    public function increment($other) { 
        $this->x += $other->x;
        $this->y += $other->y;
    }

    public function decrement($other) {
        $this->x -= $other->x;
        $this->y -= $other->y;
    }


    public function sum($other) {
        $result = $this;
        
        $result->x += $other->x;
        $result->y += $other->y;

        return $result;
    }

    public function substraction($other) {
        $result = $this;
        
        $result->x -= $other->x;
        $result->y -= $other->y;

        return $result;
    }

    public function __toString() {
            return sprintf("(%g, %g)", $this->x, $this->y);
        }
    }


$a = new Vector(1, 2);
$b = new Vector(2, 3);

echo $a;
echo $b;

?>